src/toil_scripts/gatk_germline/germline.py

```diff
@@ -1,688 +1,570 @@
+    get_shared_files.addChild(run_pipeline)
 
-    :param job: Job instance
-    :param work_dir: working directory
-    :param ids: shared file promises, dict
-    :param filenames: remaining arguments are keys for ids
 
-    :returns: shared ids dictionary
+def run_gatk_germline_pipeline(job, samples, config):
     """
-    for filename in filenames:
-        ids[filename] = job.fileStore.writeGlobalFile(os.path.join(work_dir, filename))
-    return ids
-
-
-def read_from_filestore_hc(job, work_dir, ids, *filenames):
-    """
-    Reads file from fileStore and writes it to working directory.
-
-    :param job: Job instance
-    :param work_dir: working directory
-    :param ids: shared file promises, dict
-    :param filenames: remaining arguments are filenames
-    """
-    for filename in filenames:
-        if not os.path.exists(os.path.join(work_dir, filename)):
-            job.fileStore.readGlobalFile(ids[filename], os.path.join(work_dir, filename))
-
+    Configures and runs the GATK germline pipeline with filtering.
 
-def move_to_output_dir(work_dir, output_dir, *filenames):
-    """`
-    Moves files from the working directory to the output directory.
+    VQSR is done if the run_vqsr parameter is set to True or there are at least
+    30 samples. Otherwise, each sample is hard filtered using recommended GATK parameters.
 
-    :param work_dir: the working directory
-    :param output_dir: the output directory
-    :param filenames: remaining arguments are filenames
+    :param list samples: List of tuples (uuid, url, rg_line) for joint variant calling.
+                         (i.e. samples are grouped together for VQSR). Otherwise,
+                         samples are filtered using GATK recommended hard filters.
+    :param Namespace config: Input parameters and reference files
+    :param str suffix: Adds suffix to end of filename.
+    :return: None
     """
-    for filename in filenames:
-        origin = os.path.join(work_dir, filename)
-        dest = os.path.join(output_dir, filename)
-        shutil.move(origin, dest)
+    if len(samples) == 0:
+        raise ValueError('No samples were provided!')
 
+    cores = multiprocessing.cpu_count()
 
-def batch_start(job, input_args, sample, output_dir, suffix=''):
-    """ 
-    Downloads reference data and stores fileStore promises,
-    spawns next step in pipeline - reference indexing
+    # Generate per sample gvcfs {uuid: gvcf_id}
+    gvcfs = {}
+    for uuid, url, rg_line in samples:
+        gvcfs[uuid] = job.addChildJobFn(gatk_germline_pipeline,
+                                        uuid,
+                                        url,
+                                        config,
+                                        rg_line=rg_line).rv()
 
-    :param job: Job instance
-    :param input_args: command line arguments, dict
+    # Stop after preprocessing
+    if config.preprocess_only:
+        return
```

Comment:
Sensible people will differ in their feelings here, but I would rather not do:

```
if <stopping condition>:
    return

<code to run if not stopping early>
```

Instead, I prefer:


```
if not <stopping condition>:
    <code to run if not stopping early>
```